synapse/federation/transport/client.py

```diff
@@ -68,12 +68,14 @@ async def get_room_state_ids(
         logger.debug("get_room_state_ids dest=%s, room=%s", destination, room_id)
 
         path = _create_v1_path("/state_ids/%s", room_id)
-        return await self.client.get_json(
+        resp = await self.client.get_json(
             destination,
             path=path,
             args={"event_id": event_id},
             try_trailing_slash_on_400=True,
         )
+        assert isinstance(resp, dict)
```

Comment:
This doesn't look correct. `get_json` returns whatever the remote endpoint returns; we have no way to know it is a dict. Checking that it *is* a dict shouldn't be done with assert.

The caller (`federation_client.get_room_state_ids`) has some checks that the shape of the response is correct. It should also be checking that it's a dict.

The same applies to the other assertions in this commit.